### ISSUE TITLE
False positives on reactive declarations starting from 2.11.0

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#277]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.26.0

### What version of `eslint-plugin-svelte` are you using?

2.11.0

### What did you do?

```ts
let currentTab:
    | "confirmation"
    | "destination-selection"
    | "done"
    | "introduction"
    | "moving"
    | "source-selection" = "introduction";

$: progress =
    currentTab === "introduction"
      ? 1 / 5
      : currentTab === "source-selection"
      ? 2 / 5
      : currentTab === "destination-selection"
      ? 3 / 5
      : currentTab === "confirmation"
      ? 4 / 5
      : 0;
```

### What did you expect to happen?

No errors

### What actually happened?

  9:16  error  Placing a void expression inside another expression is forbidden. Move it to its own statement instead  @typescript-eslint/no-confusing-void-expression
  9:16  error  Missing return type on function                                                                         @typescript-eslint/explicit-function-return-type
  10:5   error  Unreachable code                                                                                        no-unreachable

### Link to Minimal Reproducible Example

https://github.com/skaut/shared-drive-mover/pull/1109 - it's not minimal, I can add a minimal repro if this doesn't help...

### Additional comments

This only happens on 2.11.0, see skaut/shared-drive-mover#1109

**Comment by ota-meshi** [comment]
Thank you for reporting the issue.
It seems to parse incorrectly if there is a newline after `=`.
I will try to fix it.

**Comment by ota-meshi** [comment]
I have fixed the parser.
Could you please update svelte-eslint-parser to 0.19.1 and give it a try?

**Comment by marekdedic** [comment]
Hi,
gets even worse :(
```
Oops! Something went wrong! :(

ESLint: 8.26.0

TypeError: Cannot read properties of undefined (reading 'left')
Occurred while linting /home/user/project/src/frontend/App.svelte:47
Rule: "@typescript-eslint/no-misused-promises"
    at checkAssignment (/home/user/project/node_modules/@typescript-eslint/eslint-plugin/dist/rules/no-misused-promises.js:199:62)
    at ruleErrorHandler (/home/user/project/node_modules/eslint/lib/linter/linter.js:1115:28)
    at /home/user/project/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/home/user/project/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/home/user/project/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/home/user/project/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/home/user/project/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
    at CodePathAnalyzer.enterNode (/home/user/project/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:795:23)
    at /home/user/project/node_modules/eslint/lib/linter/linter.js:1150:32
```

**Comment by ota-meshi** [comment]
I have fixed the parser. I tried it on your repository as well and it worked.
Could you please update svelte-eslint-parser to 0.19.2 and give it a try again?

**Comment by marekdedic** [comment]
I can confirm it works now - thank you very much!
